plugins/datastreamer_plugin/ui/topic_selection_dialog/Configuration.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// This file is part of eProsima Fast DDS Monitor.
+//
+// eProsima Fast DDS Monitor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// eProsima Fast DDS Monitor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with eProsima Fast DDS Monitor. If not, see <https://www.gnu.org/licenses/>.
+
+/**
+ * @file Configuration.cpp
+ */
+
+#include "Configuration.hpp"
+#include "utils/utils.hpp"
+
+namespace eprosima {
+namespace plotjuggler {
+namespace ui {
+
+Configuration::Configuration()
+{
+}
+
+Configuration::Configuration(const QString& prefix)
+{
+    load_default_settings(prefix);
+}
+
+////////////////////
+// Save Load configuration
+bool Configuration::xmlSaveState(
+        QDomDocument& doc,
+        QDomElement& parent_element) const
+{
+    DEBUG("Saving XML configuration state");
+
+    QDomElement max_elem = doc.createElement(MAX_ARRAY_SIZE_SETTINGS_TAG);
+    max_elem.setAttribute("value", QString::number(data_type_configuration.max_array_size));
+    parent_element.appendChild(max_elem);
+
+    QDomElement stamp_elem = doc.createElement(USE_HEADER_STAMP_SETTINGS_TAG);
+    stamp_elem.setAttribute("value", data_type_configuration.use_header_stamp ? "true" : "false");
+    parent_element.appendChild(stamp_elem);
+
+    QDomElement discard_elem = doc.createElement(DISCARD_LARGE_ARRAYS_SETTINGS_TAG);
+    discard_elem.setAttribute("value", data_type_configuration.discard_large_arrays ? "true" : "false");
+    parent_element.appendChild(discard_elem);
+
+    return true;
```

Comment:
Why we need to return a boolean if always return true? Any future scenario returning false? Same for `xmlLoadState` method

Reply:
This is not perse an overriden method, but it is the equivalent to `FastDdsDataStreamer::xmlSaveState`, that it is indeed an overriden method, and it has to return a boolean.